components/post_view/combined_system_message/combined_system_message.tsx

```diff
@@ -356,19 +371,14 @@ export class CombinedSystemMessage extends React.PureComponent<Props> {
                 }
             }
 
-            if (postType === REMOVE_FROM_CHANNEL) {
-                removedUserIds.push(...userIds);
+            //Check if the postType is remove or add to manipulate a group of users.
+            if (postType === REMOVE_FROM_CHANNEL || postType === ADD_TO_CHANNEL) {
+                this.handleRemoveAndAdd(userIds, postType, content, actorId);
```

Comment:
Do we still need to handle these two cases separately? I'm not actually sure why we treated REMOVE_FROM_CHANNEL messages differently before instead of just passing it into `this.renderMessage` directly

Reply:
I actually run into problems with this .
That's why i turned the pull request into a draft, im thinking about another solution , and yes we need it to combine removed messages as when combined they don't have an actor id.
I'm still thinking about a solution ,
 The tests needs to be successful without changing anything in them

Reply:
the problem is that we can not combine removed system messages that are done by different actors this way.
user X remove user Y .
user A remove user B .
Both of the "removed" messages won't be combined unless they're done by same actor. 
@hmhealey  Do you have any solution in mind? I've tried a lot of things even taking advantage of the content array but  no luck

Reply:
Oh, I see. That makes sense for why we handled that differently before then.

What if instead of trying to group those `REMOVE_FROM_CHANNEL` regardless of actor here, we instead do that before this data gets to the component? In `combineUserActivitySystemPost`, when we're going through each post and constructing `userActivityProps`, maybe we could make it so that all `REMOVE_FROM_CHANNEL` posts are just grouped together using an empty string as an `actorId` rather than grouping them by `post.user_id`.

If you did that, you'd also have to update `extractUserActivityData` to not try to add the empty string to `allUserIds`, but I think that'd work pretty well. By the time that date gets back to `CombinedSystemMessage`, the `actorId` for all `REMOVE_FROM_CHANNEL` entries in this array will be blank, but that's fine since it's not actually used for those messages anyway.

Does that make sense? I might be missing something since I haven't touched this code in a while

Reply:
Thanks @hmhealey  ! that helped a lot and it's fixed.
I've included in the comment why I turned actorId to = "" for future devs to not get confused and waste time researching why again.
 
I had fun during this PR and  I will wait for your review !